### PR TITLE
feat: Test mid-epoch rewards reset (SC-2823)

### PR DESCRIPTION
**Pull request by deluca-mike**
# Description

# Integrations Checklist

- [ ] Have any function signatures changed? If yes, outline below.
- [ ] Have any features changed or been added? If yes, outline below.
- [ ] Have any events changed or been added? If yes, outline below.
- [ ] Has all documentation been updated?

# Changelog

### CODE REVIEW

File: contracts/core/mpl-rewards/v1/test/MplRewards.t.sol [hunk 1]
```diff
@@ -724,4 +732,77 @@ contract MplRewardsTest is DSTest {
         assertEq(rewardToken.balanceOf(address(stakers[1])),                  (dTime1_rpt + dTime2_rpt) * 30);
     }
 
+    function test_midEpochReset() external {
+        MplRewardsOwner owner      = new MplRewardsOwner();
+        SomeERC2258 rewardToken    = new SomeERC2258("RWT", "RWT");
+        SomeERC2258 stakingToken   = new SomeERC2258("SKT", "SKT");
```
Comment (lucas-manuel) [inline contracts/core/mpl-rewards/v1/test/MplRewards.t.sol]: I think RT and ST make more sense here

File: contracts/core/mpl-rewards/v1/test/MplRewards.t.sol [hunk 2]
```diff
@@ -724,4 +732,77 @@ contract MplRewardsTest is DSTest {
...
+        hevm.warp(start + ((3 * rewardsDuration) / 4));
+
+        uint256 earnedAfterLastTimeRewardApplicable = rewardsContract.earned(address(staker));
+
+        // Should not longer have been earning rewards
+        assertEq(earnedAtLastTimeRewardApplicable, earnedAfterLastTimeRewardApplicable);
+
+        // New period, take out 5000 rewards token no longer needed
+        uint256 newTotalRewardsInWad = 5_000 * WAD;
+        rewardsDuration   = 2 days;
```
Comment (lucas-manuel) [inline contracts/core/mpl-rewards/v1/test/MplRewards.t.sol]: Extra spaces

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Cosmetic changes

File: contracts/core/mpl-rewards/v1/test/MplRewards.t.sol (hunk 1 above)
Comment (deluca-mike) [reply contracts/core/mpl-rewards/v1/test/MplRewards.t.sol]: Yep. I'll fix across the tests. W not needed.

**Pull request merged**